auth0/src/test/java/com/auth0/android/request/internal/BaseRequestTest.java

```diff
@@ -124,4 +176,74 @@ public void shouldPostOnFailure() throws Exception {
         verify(callback).onFailure(eq(throwable));
         verifyNoMoreInteractions(callback);
     }
+
+    @Test
+    public void shouldBuildException() throws Exception {
+        baseRequest.onFailure(null, mock(IOException.class));
+        verify(errorBuilder).from(eq("Request failed"), any(Auth0Exception.class));
+    }
+
+    @Test
+    public void shouldParseUnsuccessfulJsonResponse() throws Exception {
+        String payload = "{key: \"value\", asd: \"123\"}";
+        final Response response = createJsonResponse(payload, 401);
+        baseRequest.parseUnsuccessfulResponse(response);
+
+        verify(errorBuilder).from(mapCaptor.capture());
+        assertThat(mapCaptor.getValue(), IsMapContaining.hasEntry("key", (Object) "value"));
+        assertThat(mapCaptor.getValue(), IsMapContaining.hasEntry("asd", (Object) "123"));
+    }
+
+    @Test
+    public void shouldParseUnsuccessfulNotJsonResponse() throws Exception {
+        String payload = "n=ot_a valid json {{]";
+        final Response response = createJsonResponse(payload, 401);
+        baseRequest.parseUnsuccessfulResponse(response);
+
+        ArgumentCaptor<String> stringCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<Integer> integerCaptor = ArgumentCaptor.forClass(Integer.class);
+        verify(errorBuilder).from(stringCaptor.capture(), integerCaptor.capture());
+        assertThat(integerCaptor.getValue(), is(401));
+        assertThat(stringCaptor.getValue(), is("n=ot_a valid json {{]"));
+    }
+
+    @Test
+    public void shouldParseUnsuccessfulInvalidResponse() throws Exception {
+        byte[] invalidBytes = new byte[]{12,23,2,1,23,3,21,3,12};
+        final Response response = createBytesResponse(invalidBytes, 401);
+        response.body().string();    //force a IOException the next time this gets called
+        baseRequest.parseUnsuccessfulResponse(response);
+
+        ArgumentCaptor<String> stringCaptor = ArgumentCaptor.forClass(String.class);
+        verify(errorBuilder).from(stringCaptor.capture(), any(Auth0Exception.class));
```

Comment:
the captor is not needed
`verify(errorBuilder).from(eq("Request to https://auth0.com/ failed"), any(Auth0Exception.class));`